unlock-app/src/components/interface/connect/ConnectedWallet.tsx

```diff
@@ -35,10 +47,39 @@ export const ConnectedWallet = () => {
   }, [signOut, deAuthenticate, setIsDisconnecting, closeConnectModal])
 
   useEffect(() => {
-    if (connected && !session && isUnlockAccount) {
+    if (connected && isUnlockAccount) {
       onSignIn()
     }
-  }, [onSignIn, connected, session, isUnlockAccount])
+  }, [onSignIn, connected, isUnlockAccount])
+
+  const {
+    register,
+    handleSubmit,
+    setError,
+    formState: { errors },
+  } = useForm<FormProps>({})
+
+  const saveEmailMutation = useMutation(async (data: FormProps) => {
+    try {
+      await storage.updateUser(data)
+      closeConnectModal()
+    } catch (error: any) {
+      if (error.response?.data?.message === 'Validation error') {
+        setError('emailAddress', {
+          message:
+            'There is already a user with that email address. Please use another one!',
+        })
+      } else {
+        setError('emailAddress', {
+          message: 'Something went wrong. Please try again later.',
+        })
+      }
+    }
+    await refetch()
```

Comment:
Need validation here and backend as well. Otherwise, I could add your email even though I do not have access to it and stop you from using yours.

Reply:
I will add validation on the "Format", but ownership validation will have to come later, since we already don't do that for Unlock accounts creation.